R/database.R

```diff
@@ -2628,3 +2636,258 @@ get_gene_expression = function(metadata,
+                                                                 min_score = min_score,
+                                                                 pass = pass,
+                                                                 with_chr_prefix = with_chr_prefix,
+                                                                 projection = projection)})
+  
+  #un-nest list into long format.
+  merged_bedpe = bind_rows(all_bedpe)
+  
+  #take out all calls that need to be lifted
+  to_be_lifted = merged_bedpe %>%
+    dplyr::filter(need_lift == TRUE)
+  
+  #lift to selected projection
+  lifted_calls = liftover_bedpe(bedpe_df = to_be_lifted, target_build = projection, print_head = FALSE)
+  
+  #subset calls that does not need a "lift"
+  no_lift_needed = merged_bedpe %>%
+    dplyr::filter(need_lift == FALSE)
+  
+  #combine calls (lifted and not lifted), arrange and sort accordingly, drop temporary column
+  merged_bedpe = rbind(lifted_calls, no_lift_needed) %>%
+    arrange(CHROM_A, CHROM_B, START_A) %>%
+    dplyr::select(-need_lift)
+  
+  #return merged manta SVs.
+  rownames(merged_bedpe) <- NULL
+  return(merged_bedpe)
+}
+
+
+#' Load the manta output (from individual flat file) for 1 sample.
+#'
+#' This function is used for retrieving Manta results (structural variants) from individual flat-files (one sample). 
+#' For multiple samples, please see get_manta_sv_by_samples (a convenience wrapper function for get_manta_by_sample). 
+#' Additional columns are extracted from the VCF column and standard filtering options are available. 
+#' This function also performs a lift-over to selected projection, if needed.   
+#'
+#' @param this_sample_id The single sample ID you want to obtain the result from.
+#' @param these_samples_metadata A metadata table containing metadata for this_sample_id, or sample of interest. This parameeter is required.
+#' @param force_lift If TRUE, coordinates will be lifted (if needed) to the selected projection. WARNING! Should always be TRUE if this function is running in stand-alone mode.
+#' @param min_vaf The minimum tumour VAF for a SV to be returned. Default value is 0.1.
+#' @param min_score The lowest Manta somatic score for a SV to be returned. Default value is 40.
+#' @param pass If set to TRUE, include SVs that are annotated with PASS in FILTER column. Default is TRUE.
+#' @param with_chr_prefix Toggle if chromosome names should be prefixed or not, default is FALSE.
+#' @param projection The projection of returned calls. Default is grch37.
+#'
+#' @return a data frame containing the Manta outputs from this_sample_id in a bedpe-like format with additional columns extracted from the VCF column.
+#' @import tidyverse
+#' @export
+#'
+#' @examples
+#' my_svs = get_manta_sv_by_sample(this_sample_id = "00-14595_tumorD",
+#'                              these_samples_metadata = get_gambl_metadata() %>% dplyr::filter(sample_id == "00-14595_tumorD"))
+#'
```

Comment:
Why is the default force_lift = TRUE ?

Reply:
I explained this in more detail in my weekly update. But anyways, it is because, **if** the user would ever want to run this function on its own, they could still get lifted coordinates back without having to change this parameter. For the `get_manta_by_samples`, this function forces this parameter to be `FALSE`, so the lifting takes place in that function instead. If that makes sense?

Reply:
I think that this may lead to liftOver being run unnecessarily though. How about setting it to FALSE by default and warning the user that if they set it to TRUE they should be sure that the file they need isn't already available from the merge (and suggest they try get_combined_sv)

Reply:
Ok, I like this. Will add this in a new commit.